### PR TITLE
CPSat solver for multiskill rcpsp

### PR DESCRIPTION
**Pull request by g-poveda**
First basic native cpsat model for multiskill problem.
- the cpsat model should be quite flexible to different variants of multiskill model :
       - variants with 1 worker per task
       - possibly only 1 skill used per worker for a given task (which is the most used mcpsp in the litterature but not really the one coded so far in do)

### CODE REVIEW

File: examples/rcpsp_multiskill/mslib/mslib_parse_and_solve.py [hunk 1]
```diff
@@ -32,5 +38,38 @@ def example_parsing_and_local_search():
...
+    files_dict = get_data_available()
+    file = [f for f in files_dict["MSLIB4"] if "MSLIB_Set4_1003.msrcp" in f][0]
+    model = parse_file_mslib(file, skill_level_version=False)
+    cp_model = CPSatMSRCPSPSolver(
+        problem=model,
+    )
+    cp_model.init_model(
+        one_worker_per_task=False, slack_skill=False, one_skill_per_task=True
+    )
+    cp_model.cp_model.Minimize(cp_model.variables["makespan"])
```
Comment (nhuet) [inline examples/rcpsp_multiskill/mslib/mslib_parse_and_solve.py:51]: Why this line? This already done in cp_model.init_model() I think ?
And if the goal was to show how to change the objective, why not use rather the higher level api: cp_model.set_model_objective("makespan")?

File: tests/rcpsp_multiskill/test_mspsp_imopse_cpsat.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+def test_imopse_cpsat():
+    file = [f for f in get_data_available() if "100_5_64_9.def" in f][0]
+    model, _ = parse_file(file, max_horizon=1000)
+    cp_model = CPSatMSRCPSPSolver(
+        problem=model,
+    )
+    cp_model.init_model(
+        one_worker_per_task=True,
+    )
+    cp_model.cp_model.Minimize(cp_model.variables["makespan"])
```
Comment (nhuet) [inline tests/rcpsp_multiskill/test_mspsp_imopse_cpsat.py:28]: Same. Why this line?

**Review by nhuet — APPROVED** [review]
Apart from the 2 lines resetting the objective outside init_model(), lgtm.

File: tests/rcpsp_multiskill/test_mspsp_imopse_cpsat.py (hunk 2 above)
Comment (g-poveda) [reply tests/rcpsp_multiskill/test_mspsp_imopse_cpsat.py:28]: it's not needed anymore but i think i was testing other objectives ^^, will maybe remove

**Pull request merged**